### ISSUE TITLE
addPassthroughCopy with symLinks not working

### ISSUE THREAD

**Issue by johanlef** [11ty/eleventy#530]
I have a submodule which contains templates, css and assets shared by multiple 11ty projects. I symlink the assets I need from the submodule into the assets directory in the root. This _assets_ folder should be copied to __site_ with `addPassthroughCopy('assets')` in my _.eleventy.js_ file. **Note:** I do not use an _.eleventyignore_ file and the sass from the submodule is processed without issues.

**Unfortunately, only the files in _assets_ which are not symlinks are copied to the __site_ directory. How can I tell eleventy to also passthrough my symlinked directories?**

```yaml
assets:
  fonts # symlink to /common/fonts
  logos # symlink to /common/logos
  scripts # normal directory
_site:
  assets:
    scripts # fonts, logos dirs are missing after build
common:
  fonts
  logos
  sass
...
```

I use eleventy v0.7.1

**Comment by zachleat** [comment]
Ah interesting. Eleventy uses `recursive-copy` for passthrough copy and it does not include symlinks by default.

https://www.npmjs.com/package/recursive-copy#usage

This will require exposing an option to enable this (or maybe configure all of the recursive copy options, not sure).

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by ckot** [comment]
this would be a nice feature.  is there any sort of hack/workaround I could use in the meantime?

**Comment by milahu** [comment]
still an issue. i use `pnpm` to manage `node_modules`, so for example
`node_modules/@fontsource/noto-sans` is a symlink to
`node_modules/.pnpm/@fontsource/noto-sans@4.2.2/node_modules/@fontsource/noto-sans`
and running `eleventyConfig.addPassthroughCopy("node_modules/@fontsource/noto-sans")` will copy zero files

**Comment by raxod502** [comment]
An easy workaround, add this to `.eleventy.js` to cause symlinks to be resolved when copying static files:

```js
const TemplatePassthrough = require("@11ty/eleventy/src/TemplatePassthrough");
const origCopy = TemplatePassthrough.prototype.copy;
TemplatePassthrough.prototype.copy = function (src, dest, copyOptions) {
  return origCopy.bind(this)(src, dest, { ...copyOptions, expand: true });
};
```

**Comment by tremby** [comment]
That looks very susceptible to code rot.

**Comment by raxod502** [comment]
I'm very open to a better solution, please let me know if you have one!

**Comment by milahu** [comment]
see #1686

**Comment by raxod502** [comment]
Ah, well, naturally, but until that gets reviewed and merged, I think we're stuck with workarounds like the one I posted.

**Comment by milahu** [comment]
> until that gets reviewed and merged, I think we're stuck

to solve this hostage situation, someone invented [patch-package](https://github.com/ds300/patch-package)
which also is unmaintained (oh the irony) but it works

sample use in my [alchi-book/patches](https://github.com/milahu/alchi/tree/master/src/alchi-book/patches)

**Comment by zachleat** [comment]
PR is merged, will ship with 2.0.0-canary.12

Thanks!
